### DISCUSSION TITLE
[Best Practice] Where should Authentication Strategies go?

### DISCUSSION THREAD

**Discussion by sokulski** [feathersjs/feathers#2278] [Q&A]
When creating a custom Authentication Strategy, where is the best place to to put those?

The [Anonymous authentication](BQJ-JIHu-p2iwOUcyKMF5) section of the docs demonstrates writing the Class right inside authentication.js, though I imagine that's only recommended for simple strategies like that one.

**Comment by robbyphillips** [comment]
I like moving authentication under the services directory and grouping things there.
```
.
├── app.ts
└── services/
    └── authentication/
        ├── hooks/
        │   └── special-auth-hook.ts
        ├── strategies/
        │   ├── anonymous-strategy.ts
        │   └── custom-strategy.ts
        └── authentication.ts
```

**Reply by sokulski** [reply]
Thank you very much. That's a very clean and intuitive way to organize strategies and hooks.

And apologies for starting out in an issue.  Thanks to @daffl for cleaning up after me.

**Reply by daffl** [reply]
That's ok, it's easy to move and I still have to highlight the new discussions better everywhere 😄
